setupSomeArtifacts.js

```diff
@@ -53,3 +54,179 @@ b
+  const fd = fs.openSync(jsFileName, "a");
+  fs.writeSync(fd, "\"");
+  for (let i = 0, i_finish = binLength - 1 | 0; i <= i_finish; i++) {
+    const c = binContent.charCodeAt(i);
+    let exit = 0;
+    if (c >= 32) {
+      if (c >= 127) {
+        if (c >= 128) {
+          fs.writeSync(fd, "\\x");
+          fs.writeSync(fd, arrayConv[(c >>> 4)]);
+          fs.writeSync(fd, arrayConv[c & 15]);
+        } else {
+          exit = 1;
+        }
+      } else if (c !== 92) {
+        if (c === /* "\"" */34) {
+          fs.writeSync(fd, "\\");
+          fs.writeSync(fd, arrayStr1[c]);
+        } else {
+          fs.writeSync(fd, arrayStr1[c]);
+        }
+      } else {
+        fs.writeSync(fd, "\\\\");
+      }
+    } else if (c >= 14) {
+      exit = 1;
+    } else {
+      switch (c) {
+        case 0 :
+            if (i === (binLength - 1 | 0) || binContent.charCodeAt(i + 1 | 0) < /* "0" */48 || binContent.charCodeAt(i + 1 | 0) > /* "9" */57) {
+              fs.writeSync(fd, "\\0");
+            } else {
+              exit = 1;
+            }
+            break;
+        case 8 :
+            fs.writeSync(fd, "\\b");
+            break;
+        case 9 :
+            fs.writeSync(fd, "\\t");
+            break;
+        case 10 :
+            fs.writeSync(fd, "\\n");
+            break;
+        case 1 :
+        case 2 :
+        case 3 :
+        case 4 :
+        case 5 :
+        case 6 :
+        case 7 :
+        case 11 :
+            exit = 1;
+            break;
+        case 12 :
+            fs.writeSync(fd, "\\f");
+            break;
+        case 13 :
+            fs.writeSync(fd, "\\r");
+            break;
+
+      }
+    }
+    if (exit === 1) {
+      fs.writeSync(fd, "\\x");
+      fs.writeSync(fd, arrayConv[(c >>> 4)]);
+      fs.writeSync(fd, arrayConv[c & 15]);
+    }
+
+  }
+  fs.writeSync(fd, "\"");
+  fs.closeSync(fd);
+  return /* () */0;
+}
+
+const compileReasonReactCmd = `bsb -clean-world -make-world`;
+const execConfig = {
+  cwd: path.join(__dirname, 'node_modules', 'reason-react'),
+  encoding: 'utf8',
+  stdio: [0, 1, 2],
+  shell: true
+};
+childProcess.execSync(compileReasonReactCmd, execConfig);
+
+const reasonReactBsDir = path.join(__dirname, 'node_modules', 'reason-react', 'lib', 'bs', 'src');
+const modules = [
+  {
+    cmi: path.join(reasonReactBsDir, 'reactDOMRe.cmi'),
+    cmj: path.join(reasonReactBsDir, 'reactDOMRe.cmj')
+  },
+  {
+    cmi: path.join(reasonReactBsDir, 'reactDOMServerRe.cmi'),
+    cmj: path.join(reasonReactBsDir, 'reactDOMServerRe.cmj'),
+  },
+  {
+    cmi: path.join(reasonReactBsDir, 'reactEventRe.cmi'),
+    cmj: path.join(reasonReactBsDir, 'reactEventRe.cmj'),
+  },
+  {
+    cmi: path.join(reasonReactBsDir, 'reasonReact.cmi'),
+    cmj: path.join(reasonReactBsDir, 'reasonReact.cmj'),
+  },
+  {
+    cmi: path.join(reasonReactBsDir, 'reasonReactOptimizedCreateClass.cmi'),
+    cmj: path.join(reasonReactBsDir, 'reasonReactOptimizedCreateClass.cmj'),
+  },
+];
```

Comment:
this could be rather shorter / more DRY as 
```
const moduleNames = ['reactDOMRe', 'reactDOMServerRe', 'reactEventRe', 'reasonReact', 'reasonReactOptimizedCreateClass'];
const modules = moduleNames.map(name => ({cmi: name + '.cmi', cmj: name + '.cmj'}))
```
don't feel too strongly though

Reply:
I like it!